tests/test_immutableset.py

```diff
@@ -209,7 +209,16 @@ def test_union(self):
         s2 = immutableset(["a", "c"])
         s3 = immutableset(["b", "c"])
         self.assertEqual(s1, s2.union(s3))
+        self.assertEqual(s1, s2.union(s3, check_top_type_matches=str))
         self.assertEqual(s1, s2 | s3)
+        # check union with plain iterable
+        self.assertEqual(s1, s2.union(["b", "c"]))
+
+        with self.assertRaises(RuntimeError):
```

Comment:
CI is failing because of
```suggestion
        with self.assertRaises(TypeError):
```